File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        int cnt = 0;

        if (number <= 0)
        {
            throw new ArgumentOutOfRangeException(nameof(number));
        }


        while (number != 1)
        {

            if (number % 2 == 0)
            {
                number /= 2;
            }
            else
            {

                number = 3 * number + 1;

            }


            ++cnt;
        }


        return cnt;


    }
}

File: zig/collatz-conjecture/collatz_conjecture.zig
pub const ComputationError = error{IllegalArgument};
pub fn steps(number: isize) anyerror!usize {
    if (number < 1) return ComputationError.IllegalArgument;
    var cnt: u32 = 0;
    var dup = number;
    while (dup != 1) {
        if (@mod(dup, 2) == 1) dup = dup * 3 + 1 else dup = @divTrunc(dup, 2);
        cnt += 1;
    }
    return cnt;
}
